mbuild/compound.py

```diff
@@ -1725,14 +1729,24 @@ def to_parmed(self, title='', residues=None):
             bond = pmd.Bond(atom_mapping[atom1], atom_mapping[atom2])
             structure.bonds.append(bond)
 
-        box = self.boundingbox
+        if box is None:
+            box = self.boundingbox
+            box_vec_max = box.maxs.tolist()
+            box_vec_min = box.mins.tolist()
+            for dim, val in enumerate(self.periodicity):
+                if val:
+                    box_vec_max[dim] = val
+                    box_vec_min[dim] = 0.0
+                if not val:
+                    box_vec_max[dim] += 0.25
+                    box_vec_min[dim] -= 0.25
+            box.mins = np.asarray(box_vec_min)
+            box.maxs = np.asarray(box_vec_max)
```

Comment:
This code is duplicated now. I think you can re-organize this so it only exists in `to_parmed`